src/sagas.js

```diff
@@ -581,6 +582,16 @@ function* fetchUSBFilter (action) {
   }
 }
 
+function* checkVmAviability (action) {
```

Comment:
Maybe this can be simplified:

```js
yield fetchSingleVm({ payload: { vmId: action.payload.vmId } })
const vm = yield select(state => state.getIn(['vms', 'vms', action.payload.vmId])
if (!vm) {
  yield put(redirectRoute({ route: '/' }))
}
```

It could save one request.